util/webhook/webhook.go

```diff
@@ -300,6 +300,20 @@ func (a *ArgoCDWebhookHandler) HandleEvent(payload interface{}) {
 			continue
 		}
 		for _, app := range filteredApps {
+			drySource := app.Spec.GetHydratorDrySource()
+			if sourceRevisionHasChanged(drySource, revision, touchedHead) && sourceUsesURL(drySource, webURL, repoRegexp) {
+				refreshPaths := path.GetAppRefreshPaths(&app)
+				if path.AppFilesHaveChanged(refreshPaths, changedFiles) {
+					namespacedAppInterface := a.appClientset.ArgoprojV1alpha1().Applications(app.ObjectMeta.Namespace)
+					log.Debugf("webhook trigger refresh app to hydrate '%s'", app.ObjectMeta.Name)
+					_, err = argo.RefreshApp(namespacedAppInterface, app.ObjectMeta.Name, v1alpha1.RefreshTypeNormal)
```

Comment:
Should we modify the `argo.RefreshApp` function to allow setting _just_ the hydrate annotation instead of the refresh annotation?

Reply:
Since the app's features are functioning properly, I think it's fine as it is.

If I understand correctly, when there’s an update (patch to k8s, and listen that from informer), messages are added to both the `appRefreshQueue` (roughly) and `appHydrateQueue`, and the subscribers to each queue check the execution conditions after receiving the messages.
So even if we make something like `argo.HydrateApp`, it shouldn't make a significant difference in how the messages are handled.

If I misunderstood something, could you correct me in a comment pleaase?
Thanks!